src/database/sqlite/utils.rs

```diff
@@ -165,6 +165,25 @@ fn migrate_v0_to_v1(conn: &mut rusqlite::Connection) -> Result<(), SqliteDbError
     Ok(())
 }
 
+// After Liana 1.0 we upgraded the schema to record whether a coin originated from an immature
+// coinbase transaction.
+fn migrate_v1_to_v2(conn: &mut rusqlite::Connection) -> Result<(), SqliteDbError> {
+    db_exec(conn, |tx| {
+        tx.execute(
+            "ALTER TABLE coins ADD COLUMN is_immature",
+            rusqlite::params![],
+        )?;
+        tx.execute(
+            "UPDATE coins SET is_immature = 0 WHERE is_immature IS NULL",
+            rusqlite::params![],
+        )?;
```

Comment:
After the update, I would alter the column to make the is_immature `not null`

Reply:
Wouldn't that mean we'd have to ALTER it also when creating it for the first time? The schema currently doesn't have `NOT NULL`.

Reply:
https://github.com/wizardsardine/liana/pull/578/files#diff-9afc8c965492b82a57677899c77edda2d3c6ec9da5a31863b3550ba3d072da88R60
I may misunderstood this line then

Reply:
oh, no, you're right. It doesn't match what i did for the first migration..

Reply:
I guess i could even go with the one liner
```sql
ALTER TABLE coins ADD COLUMN is_immature NOT NULL DEFAULT 0
```

Or maybe even with the `CHECK (is_immature IN (0,1))`.

Reply:
So i decided to not go with my approach to have exactly the same schema as if the database was freshly created (without any `DEFAULT`). But it's not possible to use your approach in SQLite: https://www3.sqlite.org/matrix/lang_altertable.html.

I'll just go with my one liner then.